opt: improve EXPLAIN for render

RaduBerinde opened the pull request:
When we EXPLAIN a render node, we show the renders in order.
Because we coalesce simple column rearrangements, this order might not
correspond to the order of the shown columns.

This change improves the output to mention the target column instead
of an index.

Release note (sql change): Minor improvement to EXPLAIN output for
"render" node.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59313)
<!-- Reviewable:end -->

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> :tada: 

> we show the renders in the order

[commmit msg nit] => "we show the renders in order"

> Reviewed 32 of 32 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59313#-:-MRgiuen4SjPSO5vsqq1:b-mftwax)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
opt: improve EXPLAIN for render

When we EXPLAIN a render node, we show the renders in order. Because
we coalesce simple column rearrangements, this order might not
correspond to the order of the shown columns.

This change improves the output to mention the target column instead
of an index.

Release note (sql change): Minor improvement to EXPLAIN output for
"render" node.

RaduBerinde commented:
Fixed, TFTR!

bors r+
